src/test/kotlin/com/contentful/java/cma/CMAHttpExceptionTest.kt

```diff
@@ -0,0 +1,232 @@
+package com.contentful.java.cma
+
+import com.contentful.java.cma.lib.TestUtils
+import com.contentful.java.cma.model.CMAHttpException
+import okhttp3.*
+import kotlin.test.assertEquals
+import org.junit.Test as test
+
+class CMAHttpExceptionTest : BaseTest() {
+    val HEADER_RATE_LIMIT_HOUR_LIMIT = "X-Contentful-RateLimit-Hour-Limit"
+    val HEADER_RATE_LIMIT_HOUR_REMAINING = "X-Contentful-RateLimit-Hour-Remaining"
+    val HEADER_RATE_LIMIT_SECOND_LIMIT = "X-Contentful-RateLimit-Second-Limit"
+    val HEADER_RATE_LIMIT_SECOND_REMAINING = "X-Contentful-RateLimit-Second-Remaining"
+    val HEADER_RATE_LIMIT_RESET = "X-Contentful-RateLimit-Reset"
+
+    @test
+    fun testHeadersParsing() {
+        val request = Request.Builder().url("https://example.com/foo").build()
+
+        val response = buildMockedResponse(request, "{\"sys\":{\"id\":\"manualid\"}}")
+
+        val e = CMAHttpException(request, response)
+
+        assertEquals("message", e.responseMessage())
+
+        assertEquals(1, e.rateLimitHourLimit())
+        assertEquals(2, e.rateLimitHourRemaining())
+        assertEquals(3, e.rateLimitSecondLimit())
+        assertEquals(4, e.rateLimitSecondRemaining())
+
+        assertEquals(-1, e.rateLimitReset(), "Error while parsing number.")
+
+        assertEquals(500, e.responseCode())
+
+        assertEquals("FAILED ErrorBody { sys = Sys { id = manualid, } }\n" +
+                "\tRequest{method=GET, url=https://example.com/foo, tag=null}\n" +
+                "\t↳ Header{}\n" +
+                "\tResponse{" +
+                "protocol=http/1.1, code=500, message=message, url=https://example.com/foo}\n" +
+                "\t↳ Header{X-Contentful-RateLimit-Hour-Limit: 1, " +
+                "X-Contentful-RateLimit-Hour-Remaining: 2, X-Contentful-RateLimit-Reset: 5k, " +
+                "X-Contentful-RateLimit-Second-Limit: 3, " +
+                "X-Contentful-RateLimit-Second-Remaining: 4}",
+                e.toString())
+    }
+
+    @test
+    fun testUnknownPath() {
+        val request = Request.Builder().url("https://example.com/foo").build()
+
+        val response = buildMockedResponse(request, TestUtils.fileToString("error_unknown_path.json"))
+
+        val e = CMAHttpException(request, response).errorBody
+
+        assertEquals(
+                "The query you sent was invalid. Probably a filter or ordering specification is not applicable to the type of a field.",
+                e.message)
+
+        assertEquals("InvalidQuery", e.sys.id)
+        assertEquals("f5841eb3c46dc4a2c4f67c2d5a027f1c", e.requestId)
+
+        assertEquals(1, e.details.errors.size)
+        assertEquals("unknown", e.details.errors[0].name)
+        val path = e.details.errors[0].path as List<*>
+        assertEquals("sys", path[0])
+        assertEquals("foo", path[1])
+        assertEquals("The path \"sys.foo\" is not recognized", e.details.errors[0].details)
+        assertEquals("f5841eb3c46dc4a2c4f67c2d5a027f1c", e.requestId)
```

Comment:
you tested this twice ;-)

Reply:
:man_facepalming: I'll be removing one of them ... :D